samcli/commands/local/lib/cfn_api_provider.py

```diff
@@ -103,3 +120,102 @@ def _extract_cloud_formation_stage(resources, stage_resource, collector):
 
         collector.stage_name = stage_name
         collector.stage_variables = stage_variables
+
+    def _extract_cloud_formation_method(self, resources, logical_id, api_resource, collector):
+        """
+        Extract APIs from AWS::ApiGateway::Method and work backwards up the tree to resolve and find the true path.
+
+        Parameters
+        ----------
+        logical_id : str
+            Logical ID of the resource
+
+        api_resource : dict
+            Resource definition, including its properties
+
+        resources: dict
+            All Resource definition, including its properties
+
+        stage_resource : dict
+            Stage Resource definition, including its properties
+
+        collector : ApiCollector
+            Instance of the API collector that where we will save the API information
+        """
+
+        properties = api_resource.get("Properties", {})
+        resource_id = properties.get("ResourceId")
+        rest_api_id = properties.get("RestApiId")
+        method = properties.get("HttpMethod")
+
+        resource_path = ""
+        if isinstance(resource_id, string_types):
+            resource = resources.get(resource_id)
```

Comment:
It would help in being more explicit about what resource this is. It took me awhile to understand what was going on here. Maybe a comment above the `if isinstance` would be helpful to.

Reply:
The if instance can be removed once we merge in the intrinsic pr since the Refs will resolve to a string.